Content.Server/Weapon/Melee/Esword/EswordSystem.cs

```diff
@@ -0,0 +1,135 @@
+using System.Collections.Generic;
+using Content.Shared.Item;
+using Content.Server.Tools.Components;
+using Content.Shared.Interaction;
+using Content.Shared.ActionBlocker;
+using Content.Shared.FixedPoint;
+using Robust.Server.GameObjects;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Audio;
+using Robust.Shared.Player;
+using Robust.Shared.IoC;
+using Robust.Shared.Maths;
+using System;
+
+namespace Content.Server.Weapon.Melee.Esword
+{
+    internal class EswordSystem : EntitySystem
+    {
+        [Dependency] private readonly ActionBlockerSystem _blockerSystem = default!;
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            SubscribeLocalEvent<EswordComponent, ComponentInit>(OnComponentInit);
+            SubscribeLocalEvent<EswordComponent, MeleeHitEvent>(OnMeleeHit);
+            SubscribeLocalEvent<EswordComponent, UseInHandEvent>(OnUseInHand);
+            SubscribeLocalEvent<EswordComponent, InteractUsingEvent>(OnInteractUsing);
+        }
+
+        private void OnComponentInit(EntityUid uid, EswordComponent comp, ComponentInit args)
+        {
+            string[] possibleColors = { "Tomato", "DodgerBlue", "Aqua", "MediumSpringGreen", "MediumOrchid" };
+            Random random = new Random();
```

Comment:
I'd probably generalise this a bit so that you can specify colours in yaml. Just give the component a datafield:
```cs
[DataField("colorOptions")]
public List<Color> ColorOptions = new()
{
    Color.Tomato,
    Color.DodgerBlue,
    Color.Aqua,
    Color.MediumSpringGreen,
    Color.MediumOrchid
};
```

Then on init, just pick from that list via
```cs
if (comp.ColorOptions.Count != 0)
    comp.BladeColor = _random.Pick(comp.ColorOptions);
```
(requires `[Dependency] private readonly IRobustRandom _random = default!;` to be added as well).